addons/account/models/account_code_mapping.py

```diff
@@ -45,14 +46,22 @@ def create(self, vals_list):
         return mappings
 
     def _search(self, domain, offset=0, limit=None, order=None) -> Query:
-        match domain:
-            case [('account_id', 'in', account_ids), *remaining_domain]:
-                return self.browse([
-                    account_id * COMPANY_OFFSET + company.id
-                    for account_id in account_ids
-                    for company in self.env.user.with_context(active_test=True).company_ids.sorted(lambda c: (c.sequence, c.name))
-                ]).filtered_domain(remaining_domain)._as_query()
-        raise NotImplementedError
+        account_ids = []
+
+        def get_accounts(condition):
+            if condition.field_expr == 'account_id' and condition.operator == 'in':
+                account_ids.extend(condition.value)
+                return Domain.TRUE
+            return condition
+
+        remaining_domain = Domain(domain).map_conditions(get_accounts)
```

Comment:
This is not correct though. For instance, the following conversion will happen:
* domain `[('account_id', 'in', [1]), ('account_id', 'in', [2])]`
* result `account.code.mapping(10001, 20001)`
* expected `account.code.mapping()`

Or
* before `['!', ('account_id', 'in', [1])]`
* result `account.code.mapping()`
* expected `account.code.mapping(20001)`


Why are we not keeping the previous implementation? I think it is better to raise than give wrong results.
Or is it because the order of the leaves is not deterministic anymore?

Reply:
`search` can be called with a Domain as an argument, so `match` won't work. Note than the previous implementation does not work with normalized domains and your second example is probably wrong (does not match the pattern).

When creating a domain, the order is preserved. However, once they go through an optimization phase, the leaves can be reordered: they are sorted (by field name) which is still a deterministic result.

I'll fix it to take the ids only of the first leaf.

Reply:
No fix then?